File: golang/344/344.go
package main

func reverseString(s string) string {
	c := []rune(s)
	for i, j := 0, len(c)-1; i < j; i, j = i+1, j-1 {
		c[i], c[j] = c[j], c[i]
	}
	return string(c)
}

File: cpp/344.cpp
// method 1: using the lib
// class Solution {
// public:
//     string reverseString(string s) {
//         reverse(s.begin(), s.end());
//         return s;
//     }
// };

// method 2: replace one by one
class Solution {
public:
    string reverseString(string s) {
       int len = s.length();
       for(int i=0, j=len-1; i<j; i++,j--){
           char t = s[i];
           s[i] = s[j];
           s[j] = t;
       }
       return s;
    }
};